ksql-rest-app/src/main/java/io/confluent/ksql/rest/entity/SourceDescription.java

```diff
@@ -111,18 +112,8 @@ public SourceDescription(
         extended,
         format,
         dataSource.getKafkaTopicName(),
-        (
-            extended && topicClient != null ? getPartitions(
-                topicClient,
-                dataSource.getKafkaTopicName()
-            ) : 0
-        ),
-        (
-            extended && topicClient != null ? getReplication(
-                topicClient,
-                dataSource.getKafkaTopicName()
-            ) : 0
-        )
+        topicDescription != null ? topicDescription.partitions().size() : 0,
```

Comment:
Is it ever null? This is only called from our code, right?  Consider either throwing an NPE or if `null` is valid, then make this _explicit_ in the interface by making it `Optional< TopicDescription>`

Reply:
It's null if the description is not extended, or if we couldn't get the description.